quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_parsing.rs

```diff
@@ -93,27 +99,25 @@ fn parse_strftime_with_tz(
 }
 
 /// Returns the appropriate tantivy datetime for the specified unix timestamp.
-pub(super) fn parse_timestamp(
-    timestamp: i64,
-    date_time_formats: &[DateTimeFormat],
-) -> Result<TantivyDateTime, String> {
-    for date_time_format in date_time_formats {
-        let date_time = match date_time_format {
-            DateTimeFormat::TimestampSecs => TantivyDateTime::from_timestamp_secs(timestamp),
-            DateTimeFormat::TimestampMillis => TantivyDateTime::from_timestamp_millis(timestamp),
-            DateTimeFormat::TimestampMicros => TantivyDateTime::from_timestamp_micros(timestamp),
-            DateTimeFormat::TimestampNanos => {
-                TantivyDateTime::from_timestamp_micros(timestamp / 1_000)
-            }
-            _ => continue,
-        };
-        return Ok(date_time);
+///
+/// This function will choose the timestamp precision based on the
+/// most significant bit position in the timestamp value.
+/// The tradeoff is that we can only support dates ranging from
+/// `01 Jan 1974 00:00:00` to `31 Dec 2241 23:59:59`
+pub(super) fn parse_timestamp(timestamp: i64) -> Result<TantivyDateTime, String> {
+    if !(MIN_TIMESTAMP_SECONDS..=MAX_TIMESTAMP_NANOSECONDS).contains(&timestamp) {
```

Comment:
I like ranges too, but in this case, I'm unsure this is more readable than:
```rust
if timestamp < MIN_TIMESTAMP_SECONDS || timestamp > MAX_TIMESTAMP_NANOSECONDS { 
```

Reply:
I had this but `cargo-clippy` advised using ranges